### DISCUSSION TITLE
Ruffle hardware acceleration

### DISCUSSION THREAD

**Discussion by Croworbit** [ruffle-rs/ruffle#12380] [General] [answered]
Does Ruffle use my hardware for games? I have a 3090 and I don't see much utilization
lots of games that have physics just run really bad, and I don't understand why
with the power I have, I think it would make sense that those types of games would take advantage of my card, and be better than on Flash player, for example which runs way better in these specific games

**Comment by Lord-McSweeney** [comment, marked as answer]
Ruffle runs almost its entire rendering on the GPU. Physics is not a GPU-side thing, though. Ruffle is also not very well-optimized for physics-simulating games (e.g. the Box2D engine). Flash has many optimizations, which allow those games to run much faster than in Ruffle. The GPU is not responsible for any of the physics stuff.
